src/Controller/AbsenceController.php

```diff
@@ -341,42 +595,663 @@ private function save(Request $request) {
+            $p->fetch();
+            $agents_tous = $p->elements;
+
+            // Tous les agents
+            foreach ($agents_tous as $elem) {
+                if (in_array($elem['id'], $perso_ids)) {
+                    $agents_selectionnes[$elem['id']] = $elem;
+                }
+            }
+
+            // Tous les agents concernés (ajoutés, supprimés, restants)
+            $agents_concernes=array();
+            // Ajoute au tableau $agents_concernes les agents qui étaient présents avant la modification
+            foreach ($agents as $elem) {
+                if (!array_key_exists($elem['perso_id'], $agents_concernes)) {
+                    $agents_concernes[$elem['perso_id']] = $agents_tous[$elem['perso_id']];
+                }
+            }
+
+            // Ajoute au tableau $agents_concernes les agents sélectionnés
+            foreach ($agents_selectionnes as $elem) {
+                if (!array_key_exists($elem['id'], $agents_concernes)) {
+                    $agents_concernes[$elem['id']] = $elem;
+                }
+            }
+
+            // Les agents supprimés de l'absence
+            $agents_supprimes=array();
+            foreach ($agents as $elem) {
+                if (!array_key_exists($elem['perso_id'], $agents_selectionnes)) {
+                    $agents_supprimes[$elem['perso_id']] = $agents_tous[$elem['perso_id']];
+                }
+            }
+
+            // Les agents ajoutés à l'absence
+            $agents_ajoutes=array();
+            foreach ($agents_selectionnes as $elem) {
+                if (!in_array($elem['id'], $perso_ids1)) {
+                    $agents_ajoutes[]=$elem;
+                }
+            }
+
+            // Comparaison des anciennes données et des nouvelles
+            $modification = (
+                !empty($agents_ajoutes)
+                or !empty($agents_supprimes)
+                or $debut1 != $debut_sql
+                or $fin1 != $fin_sql
+                or htmlentities($motif1, ENT_QUOTES|ENT_IGNORE, 'UTF-8', false) != htmlentities($motif, ENT_QUOTES|ENT_IGNORE, 'UTF-8', false)
+                or htmlentities($motif_autre1, ENT_QUOTES|ENT_IGNORE, 'UTF-8', false) != htmlentities($motif_autre, ENT_QUOTES|ENT_IGNORE, 'UTF-8', false)
+                or htmlentities($commentaires1, ENT_QUOTES|ENT_IGNORE, 'UTF-8', false) != htmlentities($commentaires, ENT_QUOTES|ENT_IGNORE, 'UTF-8', false) 
+                or $valide1 != $valide
+                or $rrule1 != $rrule
+                or empty($pj1_1) != empty($pj1)
+                or empty($pj2_1) != empty($pj2)
+                or empty($so_1) != empty($so)
+            );
+
+            // Si aucune modification, on retourne directement à la liste des absences
+            if (!$modification) {
+                $msg = urlencode("L'absence a été modifiée avec succès");
+                echo "<script type='text/JavaScript'>document.location.href='/holiday?msg=$msg&msgType=success';</script>\n";
```

Comment:
holiday => absence
Please don't echo in controllers

Reply:
I did forget this echo,  I am editing the file